### DISCUSSION TITLE
System.Diagnostics.DiagnosticSource

### DISCUSSION THREAD

**Discussion by C0NN0RAD0** [Azure/azure-functions-dotnet-worker#2275] [Feedback]
Just to let you guys know in Functions Dotnet V6 I am having nothing but issues somewhere with the assembly or incompatability with azure functions or dependencies leaving me this error : `For detailed output, run func with --verbose flag.
[2024-02-08T17:47:31.771Z] Host lock lease acquired by instance ID '000000000000000000000000EA2C7E34'.
[2024-02-08T17:48:41.653Z] Executing 'DeletePatient' (Reason='This function was programmatically called via the host APIs.', Id=b06115ff-8c2e-47de-8022-c6ba5efa99c0)
[2024-02-08T17:48:41.907Z] Error deleting patient with ID 2: Could not load file or assembly 'System.Diagnostics.DiagnosticSource, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
[2024-02-08T17:48:41.907Z] 
[2024-02-08T17:48:41.925Z] Executed 'DeletePatient' (Succeeded, Id=b06115ff-8c2e-47de-8022-c6ba5efa99c0, Duration=347ms)
` From this code:
`using System;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using System.Threading.Tasks;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using MySql.Data.MySqlClient; // Import the MySqlClient namespace instead of System.Data.SqlClient

namespace PatientDELETE
{
    public static class DeletePatientFunction
    {
        [FunctionName("DeletePatient")]
        public static async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Function, "delete", Route = "patients/{id}")] HttpRequest req,
            ILogger log,
            int id)
        {
            try
            {
                string connectionString = Environment.GetEnvironmentVariable("MySqlConnectionString"); // Update the connection string key as per your configuration

                using (MySqlConnection conn = new MySqlConnection(connectionString)) // Use MySqlConnection instead of SqlConnection
                {
                    await conn.OpenAsync();
                    var text = "DELETE FROM Patients WHERE patient_id = @Id"; // Remove "dbo." from table name

                    using (MySqlCommand cmd = new MySqlCommand(text, conn)) // Use MySqlCommand instead of SqlCommand
                    {
                        cmd.Parameters.AddWithValue("@Id", id);

                        var result = await cmd.ExecuteNonQueryAsync();
                        if (result > 0)
                        {
                            return new OkObjectResult(new { Message = $"Patient with ID {id} deleted successfully." }); // Return a message with ID of the deleted patient
                        }
                        else
                        {
                            return new NotFoundObjectResult(new { Message = $"Patient with ID {id} not found or could not be deleted." }); // Return a message with ID of the patient not found or not deleted
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                log.LogError($"Error deleting patient with ID {id}: {ex.Message}");
                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
            }
        }
    }
}
`
Hope this might help or be a known issue?

**Comment by kshyju** [comment]
From the code shared above, it looks like an in-proc dotnet function. For in-proc issues, https://github.com/Azure/azure-functions-host is the repo to open an issue against.

Keep in mind that, the current(V4) in-proc model is built on .NET6. So if you are using dependencies greater than 6.X, that may cause issues. The isolated model (the code present in this repo) is more flexible and allows you to use newer version of dependencies/.NET runtimes without issues.

See https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide?tabs=windows#benefits-of-the-isolated-worker-model
